packages/telemetry/test/detectors/circleCiDetectorSync.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { expect } from 'chai'
+
+process.env.CIRCLECI
+
+import { circleCiDetectorSync } from '../../src/detectors/circleCiDetectorSync'
+
+describe('circleCiDetectorSync', () => {
+  describe('undefined values', () => {
+    const processValues: any = {}
+
+    beforeEach(() => {
+      // cache values
+      processValues.CIRCLECI = process.env.CIRCLECI
+      processValues.CIRCLE_BRANCH = process.env.CIRCLE_BRANCH
+      processValues.CIRCLE_JOB = process.env.CIRCLE_JOB
+      processValues.CIRCLE_NODE_INDEX = process.env.CIRCLE_NODE_INDEX
+      processValues.CIRCLE_BUILD_URL = process.env.CIRCLE_BUILD_URL
+      processValues.CIRCLE_BUILD_NUM = process.env.CIRCLE_BUILD_NUM
+      processValues.CIRCLE_SHA1 = process.env.CIRCLE_SHA1
+      processValues.CIRCLE_PR_NUMBER = process.env.CIRCLE_PR_NUMBER
+
+      //reset values
+      delete process.env.CIRCLECI
+      delete process.env.CIRCLE_BRANCH
+      delete process.env.CIRCLE_JOB
+      delete process.env.CIRCLE_NODE_INDEX
+      delete process.env.CIRCLE_BUILD_URL
+      delete process.env.CIRCLE_BUILD_NUM
+      delete process.env.CIRCLE_SHA1
+      delete process.env.CIRCLE_PR_NUMBER
+    })
+
+    afterEach(() => {
+      // Replace values
+      process.env.CIRCLECI = processValues.CIRCLECI
```

Comment:
Does this condition need to be reversed so we delete the circle process variables so they don't leak out to other tests in the process?

Reply:
this is the reversing, we cache the values, delete them, then reset them to the cache

Reply:
Ah that makes a lot of sense now 😅